sse_client.js

```diff
@@ -18,130 +18,132 @@ function isEmpty(obj) {
                     node.timerId = null;
                 }
             }
-            
+
             // Check whether the stream should be paused
             if (msg.pause === true) {
-                node.status({fill: 'yellow', shape: 'ring', text: 'paused'});
+                node.status({ fill: 'yellow', shape: 'ring', text: 'paused' });
                 node.paused = true;
-                
+
                 return;
             }
-            
+
             // To stop the streaming, close the client
             if (msg.stop === true) {
-                node.status({fill: 'red', shape: 'ring', text: 'disconnected'});
+                node.status({ fill: 'red', shape: 'ring', text: 'disconnected' });
                 if (node.client) {
                     node.client.close();
                 }
                 node.client = null;
                 return;
             }
-            
+
             // When the previous client is NOT paused, we will stop it and create a new one (to send a new http GET).
             if (node.client && !node.paused) {
-                node.status({fill: 'red', shape: 'ring', text: 'disconnected'});
+                setTimeout(() => {
+                    node.status({ fill: 'red', shape: 'ring', text: 'disconnected' });
+                }, 1000); // 08/04/2022 Supergiovane: Allow the "reconnecting" status to be shown, in case of reconnection by timeout.                
                 node.client.close();
                 node.client = null;
             }
-            
+
             // When we arrive here, a new stream should be started or a paused stream should be restarted
-            
+
             // When the previous client is paused, then resume it again
             if (node.client && node.paused) {
-                node.status({fill: "green", shape: "dot", text: "connected"});
-                node.paused = false; 
-                
-                // When a timeout is specified, start a new timer (that restarts the SSE client)
-                if (node.restart) {
-                    node.timerId = setTimeout(function restartClient() {
-                        // Restart the SSE client by resending the last message again
-                        handleMsg(node.prevMsg);
-                    }, node.timeout * 1000);
-                }
+                node.status({ fill: "green", shape: "dot", text: "connected" });
+                node.paused = false;
             }
-                
+
+            // When a timeout is specified, start a new timer (that restarts the SSE client)
+            if (node.restart) startTimeoutTimer();
+
             // When no client is available, just create one (and connect to the server) ...  
             if (!node.client) {
                 // All EventSource parameter should be passed to the constructor as a dictionary
-                var options = {};
-                var url = '';
+                let options = {};
```

Comment:
Just out of curiosity: was this (var instead of let) causing the memory leak?

Reply:
Hi Bart
That "let" contributes to have a lesser memory footprint, thus helping the garbace collector to free them.
Scoping variables with "let", let the variable ceasing to exist as soon as the function block finishes living, so the garbace collector can free such unreferenced memory.
Scoping it with "var", instead, let the variable live for the entire node lifecycle and reinstantiating it everytime could lead in mem leaking.